lang/c/build.sh

```diff
@@ -20,84 +20,82 @@
 
 set -e        # exit on error
 
+cd "$(dirname "$0")" # If being called from another folder, cd into the directory containing this script.
+
+# shellcheck disable=SC1091
+source ../../share/build-helper.sh "C"
+
 root_dir=$(pwd)
-build_dir="../../build/c"
-dist_dir="../../dist/c"
+build_dir="$BUILD_ROOT/build/c"
+dist_dir="$BUILD_ROOT/dist/c"
 version=$(./version.sh project)
 tarball="avro-c-$version.tar.gz"
-doc_dir="../../build/avro-doc-$version/api/c"
+doc_dir="$BUILD_ROOT/build/avro-doc-$version/api/c"
 
-function prepare_build {
-  clean
-  mkdir -p $build_dir
-  (cd $build_dir && cmake $root_dir -DCMAKE_BUILD_TYPE=RelWithDebInfo)
+function prepare_build()
+{
+  command_clean
+  execute mkdir -p "$build_dir"
```

Comment:
Where is `execute` defined?

Reply:
share/build-helper.sh in the PR. The main reason behind the `execute` concept was to be able to have a `--dry-run` option, where the build.sh just prints out what would be running. The user can copy paste it and run it manually step by step if needed, or just simply checking the steps.